Increase timeout for the PPD file to be copied so that it doesn't fail
on low-end configurations.

Last chunk: fix for CVE-2024-47175.

Index: scheduler/ipp.c
--- scheduler/ipp.c.orig
+++ scheduler/ipp.c
@@ -4492,7 +4492,7 @@ copy_model(cupsd_client_t *con,		/* I - Client connect
   close(temppipe[1]);
 
  /*
-  * Wait up to 30 seconds for the PPD file to be copied...
+  * Wait up to 90 seconds for the PPD file to be copied...
   */
 
   total = 0;
@@ -4512,7 +4512,7 @@ copy_model(cupsd_client_t *con,		/* I - Client connect
     FD_SET(temppipe[0], &input);
     FD_SET(CGIPipes[0], &input);
 
-    timeout.tv_sec  = 30;
+    timeout.tv_sec  = 90;
     timeout.tv_usec = 0;
 
     if ((i = select(maxfd, &input, NULL, NULL, &timeout)) < 0)
@@ -5415,6 +5415,13 @@ create_local_bg_thread(
       }
       ippDelete(response2);
     }
+  }
+
+  // Validate response from printer...
+  if (!ippValidateAttributes(response))
+  {
+    send_ipp_status(con, IPP_STATUS_ERROR_DEVICE, _("Printer returned invalid data: %s"), cupsLastErrorString());
+    goto finish_response;
   }
 
   // TODO: Grab printer icon file...
